front/components/items/Appointment.tsx

```diff
@@ -53,22 +57,28 @@ const Appointment = () => {
     }
     const allDates: string[] | undefined = appointmentsTherapist?.map(item => item.attributes.date);
     const stringDate: string[] | undefined = allDates?.filter((item, index) => {return allDates?.indexOf(item) === index});
+    // deleting repeated dates 
+    /* indexOf is searching the first index numer of current item value and 
+    then compering to the left elements. If they're same, value's true, 
+    if they're reapeted value's false and won't stay in new array
+    e.g "water" is 0, but also 3, 4, 5, so only "0 water" would stay, because 0 doesn't equal 3 */
     const availableDate: Date[] | undefined = stringDate?.map(item => new Date(item));
     let allHours: Date[] | undefined = appointmentsDate?.map(item => item.attributes.time);
     let sortHours: Date[] | undefined = allHours?.sort(function (a, b) {
```

Comment:
in code you have Number - Number, here you want to have Date. You need to convert Numbers to date, or dont use this type